backend_plugin/src/evaluators/EvaluatorManager.java

```diff
@@ -94,4 +72,41 @@ public void addEvaluator(IEditorPart editorWindow) {
 	public void notifyFeatureSuggestion(String featureID) {
 		this.fs.notifyAllObservers(featureID);
 	}
+
+	public void start() {
+		for (IWorkbenchWindow window : PlatformUI.getWorkbench().getWorkbenchWindows()) {
+			for (IWorkbenchPage page : window.getPages()) {
+				EditorWindowListener windowListener = new EditorWindowListener(this);
+				page.addPartListener(windowListener);
+				this.openWindowListeners.put(page, windowListener);
+				for (IEditorReference editRef : page.getEditorReferences()) {
+					IEditorPart ePart = editRef.getEditor(false);
+					if (ePart != null) {
+						IEditorInput eInput = ePart.getEditorInput();
+						if (eInput != null) {
+							String filename = eInput.getName();
+							if (filename != null && filename.endsWith(".java") &&
+									!this.getOpenEvaluators().containsKey(ePart)) {
+								addEvaluator(ePart);
+							}
+						}
+					}
+				}
+			}
+		}
+	}
+
+	/**
+	 * Stops all Evaluators and removes any window or document listeners
+	 */
+	public void stop() {
+		for (IWorkbenchPage workbenchPage : this.openWindowListeners.keySet()) {
```

Comment:
Same here in the stop method.  Something like "cycle through all workbenches and remove any listeners".